corehq/apps/linked_domain/local_accessors.py

```diff
@@ -48,3 +49,25 @@ def _to_json(role):
         return _clean_json(role.to_json())
 
     return [_to_json(role) for role in UserRole.by_domain(domain)]
+
+
+def get_data_dictionary(domain):
+    data_dictionary = {}
+    for case_type in CaseType.objects.filter(domain=domain):
```

Comment:
It sounds safe, in that I don't think it will cause problems. It will just be missing some case properties (which are also missing from the master). I don't think the behavior would be surprising (it will match the master at the time you pushed), although if generation of the master is in progress, they may quickly diverge. I think that is kind of wishy washy but I lean toward this being fine, although see both sides of the issue.